pkl/private/pkl_codegen_java.bzl

```diff
@@ -134,14 +128,14 @@ def pkl_config_java_library(name, files, module_path = [], generate_getters = No
     # Ensure that there are no duplicate entries in the deps
     all_deps = depset(
         pkl_deps + module_path,
-        transitive = [depset([dep]) for dep in deps],
+        transitive = [depset([dep]) for dep in deps if PklFileInfo in dep or PklCacheInfo in dep],
```

Comment:
Can we error out instead if neither provider is set in at least one element of `deps`?